frontend/components/Corpus.js

```diff
@@ -0,0 +1,147 @@
+import React, {useEffect, useState} from "react";
+import * as PropTypes from "prop-types";
+// import STYLES from "./Corpus.module.scss";
+import {getCookie} from "../common";
+import {Modal} from "react-bootstrap";
+
+const Corpus = ({id}) => {
+
+    const [corpusData, setCorpusData] = useState({});
+    const [allDocData, setAllDocData] = useState([]);
+    const [containsDoc, setContainsDoc] = useState({});
+    const [loading, setLoading] = useState(true);
+    const [showModal, setShowModal] = useState(false);
+    const handleOpenModal = () => setShowModal(true);
+    const handleCloseModal = () => setShowModal(false);
+
+    useEffect(() => {
+        fetch(`/api/corpus/${id}`)
+            .then(response => response.json())
+            .then(data => {
+                setCorpusData(data);
+                fetch("/api/all_documents")
```

Comment:
I'm guessing this is a temporary measure to help you all progress through this part of the UI, but I'd recommend we create an endpoint for retrieving only a subset of `all_documents` based on the `Corpus.documents` list and returning whatever useful information we need to display here (probably the same set of things we display on the documents page). If we imagine a future where we have thousands of `Document` objects in the database, this would require anyone who accesses any `Corpus` to retrieve some data representation of all of those and then filter them on the front end. That'll never be as efficient as letting the database query do that sorting.

Reply:
We currently call `/api/all_documents` because we wanted to display all the `documents` in the Modal to allow users to update which documents they want to include in the corpus. Should we only call `/api/all_documents` when we want to update the documents in the corpus or should we look into another way of allowing users to update the documents in the corpus?

Reply:
Ah, thanks for the clarification. That makes total sense. In the future, we'd likely want some kind of searching mechanism (i.e., find document by title or something similar). For now, retrieving a simplified view of all documents makes sense for the use case.